augur/tasks/github/issues/tasks.py

```diff
@@ -129,19 +130,20 @@ def process_issues(issues, task_name, repo_id, logger) -> None:
 
         # insert contributors from these issues
         logger.info(f"{task_name}: Inserting {len(contributors)} contributors")
-        session.insert_data(contributors, Contributor, ["cntrb_login"])
+        session.insert_data(contributors, Contributor, ["cntrb_id"])
                             
 
         # insert the issues into the issues table. 
         # issue_urls are gloablly unique across github so we are using it to determine whether an issue we collected is already in the table
         # specified in issue_return_columns is the columns of data we want returned. This data will return in this form; {"issue_url": url, "issue_id": id}
         logger.info(f"{task_name}: Inserting {len(issue_dicts)} issues")
-        issue_natural_keys = ["repo_id", "gh_issue_id"]
+        issue_natural_keys = ["issue_url"]
         issue_return_columns = ["issue_url", "issue_id"]
         issue_string_columns = ["issue_title", "issue_body"]
-        issue_return_data = session.insert_data(issue_dicts, Issue, issue_natural_keys, return_columns=issue_return_columns, string_fields=issue_string_columns)
-
-
+        try:
```

Comment:
@sgoggins I would like to change this to throw an exception when in development but not throw an exception during production. We can use @Ulincsys dev environment variable.  The only other option than that is to throw an exception in production and have people open issues when they get exceptions, but I don't think we want that.

Reply:
@ABrain7710 : If we do not throw an exception does it simply not update the issue and move ahead? I will trust your judgement on this one. Do what you think should be done.

Reply:
@sgoggins Yes, if we catch the exception, it will fail to insert the row, but the process will keep running. So, the issue assignees and labels will not be inserted, and the task will finish. If we do this for production instances and then raise the errors in development so we can fix them, it causes two things. First is a lot less user testing of the code because the errors are being squashed, and they are never notified of it, which means we are not notified. Second, we have to test a lot more because we won't get as many issues from people.

I'm not saying I have a problem with this approach. This is actually the approach I like the most. I'm just stating that this requires us to do a lot more large-scale testing to find the weird bugs.

Reply:
@sgoggins This is what I'm proposing.
```
        try:
            issue_return_data = session.insert_data(issue_dicts, Issue, issue_natural_keys, return_columns=issue_return_columns, string_fields=issue_string_columns)
        except IntegrityError as e:
            logger.error(f"Ran into integrity error:{e} \n Offending data: \n{issue_dicts}")

            if development:
                raise e
```